kairon/api/app/routers/bot.py

```diff
@@ -803,3 +803,70 @@ async def validate_training_data(
                               current_user.get_bot(), current_user.get_user(),
                               False, False)
     return {"message": "Event triggered! Check logs."}
+
+
+@router.get("/EntitySynonyms", response_model=Response)
+async def get_all_synonyms(
+        current_user: User = Depends(auth.get_current_user_and_bot),
+):
+    """
+    Fetches the stored synonyms of the bot
+    """
+    synonyms = list(mongo_processor.fetch_synonyms(current_user.get_bot()))
+    return {"data": synonyms}
+
+
+@router.post("/EntitySynonyms", response_model=Response)
+async def add_synonyms(
+        request_data: SynonymRequest,
+        current_user: User = Depends(auth.get_current_user_and_bot)
+):
+    """
+    adds a new synonym and its values
+    :param request_data:
+    :param current_user:
+    :return: Success message
+    """
+    try:
+        synonym_value = request_data.dict()
+        mongo_processor.add_synonym(synonyms_dict=synonym_value, bot=current_user.get_bot(), user=current_user.get_user())
+    except AppException as ae:
```

Comment:
either remove unnecessory exception or add test where exception is thrown by add method